packages/demo/src/demo-building-blocs/PropsDoc.tsx

```diff
@@ -0,0 +1,101 @@
+import {createSystem, createVirtualTypeScriptEnvironment} from '@typescript/vfs';
+import * as React from 'react';
+import {Loading} from 'react-vapor';
+import * as ts from 'typescript';
+import '@demo-styling/props-doc.scss';
+
+import {useTypescriptServer} from './useTypescriptServer';
+
+interface PropInfo {
+    name: string;
+    description: string;
+    type: string;
+    optional: boolean;
+}
+
+export const PropsDoc: React.FunctionComponent<{componentName: string}> = ({componentName}) => {
+    const [propsList, setPropsList] = React.useState<PropInfo[]>();
+    const {fsMap, compilerOptions} = useTypescriptServer();
+
+    React.useEffect(() => {
+        if (fsMap === null) {
+            return;
+        }
+        const system = createSystem(fsMap);
+        const content = `import * as React from 'react';
+import {${componentName}} from 'react-vapor';
+const props: React.ComponentProps<typeof ${componentName}> = {`;
```

Comment:
```suggestion
        const content = `import * as React from 'react';
                                  import {${componentName}} from 'react-vapor';
                                  const props: React.ComponentProps<typeof ${componentName}> = {`;
```

Reply:
This would create useless whitespaces in the string

Reply:
oh i see  :eyes: